Add support for confidential VMs (#3265)
kubernetes-sigs/cluster-api-provider-azure: docs/book/src/topics/trusted-launch-for-vms.md

# Trusted launch for VMs

This document describes how to deploy a cluster with nodes that support [trusted launch](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch).

## Limitations

Before you begin, be aware of the following:

- [Limitations](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#limitations)
- [SecureBoot](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#secure-boot)
- [vTPM](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#vtpm)

## Trusted Launch Images

> One of the limitations of trusted launch for VMs is that they are supported by [generation 2](https://learn.microsoft.com/en-us/azure/virtual-machines/generation-2) VMs.

Reviewer: ```suggestion
One of the limitations of trusted launch for VMs is that they require [generation 2](https://learn.microsoft.com/en-us/azure/virtual-machines/generation-2) VMs.
```

Trusted launch supported OS images are not included in the list of `capi` reference images. Before creating a cluster hosted on VMs with trusted launch features enabled, you can create a [custom image](custom-images.md) based on a one of the trusted launch supported OS images using [image-builder](https://github.com/kubernetes-sigs/image-builder). For example, you can run the following to create such an image based on Ubuntu Server 22.04 LTS:

```bash

> $ make -C build-azure-sig-ubuntu-2204-gen2

Reviewer: ```suggestion
$ make -C images/capi build-azure-sig-ubuntu-2204-gen2
```
(Or else drop the `-C`, assuming you already did `cd images/capi` in the image-builder project.)
